lucene/core/src/java/org/apache/lucene/index/BufferedUpdates.java

```diff
@@ -288,15 +186,24 @@ void clear() {
     deleteTerms.clear();
     deleteQueries.clear();
     deleteDocIDs.clear();
-    numericUpdates.clear();
-    binaryUpdates.clear();
     numTermDeletes.set(0);
-    numNumericUpdates.set(0);
-    numBinaryUpdates.set(0);
-    bytesUsed.set(0);
+    numFieldUpdates.set(0);
+    fieldUpdates.clear();
+    bytesUsed.addAndGet(-bytesUsed.get());
```

Comment:
Unrelated to this PR, but it feels like `Counter` could expose a `reset()` method and internally set the value to 0, instead of everyone doing this juggling.

Reply:
agreed I will open a followup.